crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -75,21 +86,47 @@ pub(crate) fn parenthesize_chained_logical_operators(
                     ..
                 },
             ) => {
+                let locator = checker.locator();
+                let source_range = bool_op.range();
                 if parenthesized_range(
                     bool_op.into(),
                     expr.into(),
                     checker.indexer().comment_ranges(),
-                    checker.locator().contents(),
+                    locator.contents(),
                 )
                 .is_none()
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        ParenthesizeChainedOperators,
-                        bool_op.range(),
-                    ));
+                    let mut diagnostic =
+                        Diagnostic::new(ParenthesizeChainedOperators, source_range);
+                    if is_single_line_expr_with_narrow_width(
+                        locator,
+                        source_range,
+                        checker.settings,
+                    ) {
+                        let new_source = format!("({})", locator.slice(source_range));
+                        diagnostic.set_fix(Fix::applicable_edit(
+                            Edit::range_replacement(new_source, source_range),
+                            Applicability::Safe,
+                        ));
+                    }
+                    checker.diagnostics.push(diagnostic);
                 }
             }
             _ => continue,
         };
     }
 }
+
+fn is_single_line_expr_with_narrow_width(
+    locator: &Locator,
+    source_range: TextRange,
+    settings: &LinterSettings,
+) -> bool {
+    if locator.contains_line_break(source_range) {
+        return false;
+    }
```

Comment:
Can you share more on the motivation behind this limitation?

Reply:
See https://github.com/astral-sh/ruff/pull/9449#discussion_r1446083411